ucm2/Intel/sof-hda-dsp/HiFi-sof.conf

```diff
@@ -0,0 +1,189 @@
+	      HeadphoneDrcBlob "${var:BlobPath}/drc/passthrough.blob"
+	}
+}
+
+Define.ConfPathFromDMI "${sys:devices/virtual/dmi/id/sys_vendor}/${sys:devices/virtual/dmi/id/product_name}.conf"
+Define.SOFProductConfig "/sof/product_configs/${var:ConfPathFromDMI}"
+Define.SOFConfFullPath "${ConfTopDir}${var:SOFProductConfig}"
+Define.SOFUserConfig "/sof/user_configs/${var:ConfPathFromDMI}"
+Define.SOFUserConfFullPath "${ConfTopDir}${var:SOFUserConfig}"
+
+If.HasUserConfig {
+	Condition {
+		Type Path
+		Mode read
+		Path "${var:SOFUserConfFullPath}"
+	}
+	True {
+		Include.product-config.File "${var:SOFUserConfig}"
+	}
+	False.If.HasProductConfig {
+		Condition {
+			Type Path
+			Mode read
+			Path "${var:SOFConfFullPath}"
+		}
+		True {
+			Include.product-config.File "${var:SOFProductConfig}"
+		}
+	}
+}
+
+# Redefine headphone control FOR SOF. This is same as Headphone in HDA/HiFi-analog.conf
+# but adds DRC and EQ controls.
+
+DefineMacro.SofControl {
+        Define.EndpointDrcVar "${var:__endpoint}DrcBlob"
+        Define.EndpointIirVar "${var:__endpoint}IirBlob"
+        Define.EndpointFirVar "${var:__endpoint}FirBlob"
+        Define.EndpointDrcBlob "${var:$EndpointDrcVar}"
+        Define.EndpointIirBlob "${var:$EndpointIirVar}"
+        Define.EndpointFirBlob "${var:$EndpointFirVar}"
+
+       If.endpoint_with_drc {
+                Condition {
+                        Type ControlExists
+                        Control "name='${var:PostMixerAnalogPlaybackDRCSwitch}'"
+                }
+                True {
+                        EnableSequence [
+                                cset "name='${var:PostMixerAnalogPlaybackDRCSwitch}' ${var:__drcswitch}"
+                                cset-tlv "name='${var:PostMixerAnalogPlaybackDRCBytes}' ${var:EndpointDrcBlob}"
+                                cset-tlv "name='${var:PostMixerAnalogPlaybackIIRBytes}' ${var:EndpointIirBlob}"
+                                cset-tlv "name='${var:PostMixerAnalogPlaybackFIRBytes}' ${var:EndpointFirBlob}"
+                                #shell "/bin/echo '${var:__endpoint} ${var:EndpointIirBlob} ${var:EndpointFirBlob} ${var:EndpointDrcBlob}' >> /tmp/alsa-ucm.txt"
+                        ]
+                }
+                False.If.endpoint_with_eq {
+                        Condition {
+                                Type ControlExists
+                                Control "name='${var:PostMixerAnalogPlaybackFIRBytes}'"
+                        }
+                        True {
+                                EnableSequence [
+                                        cset-tlv "name='${var:PostMixerAnalogPlaybackIIRBytes}' ${var:EndpointIirBlob}"
+                                        cset-tlv "name='${var:PostMixerAnalogPlaybackFIRBytes}' ${var:EndpointFirBlob}"
+                                        #shell "/bin/echo '${var:__endpoint} ${var:EndpointIirBlob} ${var:EndpointFirBlob}' >> /tmp/alsa-ucm.txt"
```

Comment:
I know it's not common to leave such commented-out code, but I did find these immensely useful when debugging and I could imagine these are useful for anyone who wants to add tuning for new laptop configurations later one. So if possible, +1 to leave these as examples.

Reply:
I moved the commented shell up before controls set to see the print before fails with them. There's now also a comment about use for debugging.